### PR TITLE
PropertyGraph set index to vertex and edge ids

### LINKED ISSUE #2401: [FEA] Set index to `_EDGE_ID_ ` and `_VERTEX_` for `_vertex_prop_dataframe` and `_edge_prop_dataframe`  to make sampling faster
**Describe the solution you'd like and any additional context**

We should set index to `_EDGE_ID_` and `_VERTEX_`  for `_vertex_prop_dataframe` and `_edge_prop_dataframe`  so that when we are fetching for sampling by `ids` we are fast. 

Motivating Example where we see a 3x speed up for fetching a batch of `50k`. 

```python
from cugraph.experimental import PropertyGraph
import numpy as np
import cudf

pg = PropertyGraph()
n_features = 100
n_rows = 10_000_000

df = cudf.DataFrame({'node_id':np.arange(n_rows)})
for feat_id in range(n_features):
    df[f'feat_{feat_id}'] = np.ones(n_rows)
pg.add_vertex_data(df,vertex_col_name='node_id')


node_ids_to_fetch = np.random.randint(100_000_000, size=50_000)

```

Without  Index:
```python
%%timeit
node_ids_df = cudf.DataFrame({'_VERTEX_':node_ids_to_fetch, 'input_order':np.arange(0,len(node_ids_to_fetch))})
fetched_df = node_ids_df.merge(pg._vertex_prop_dataframe, how='left')
fetched_df = fetched_df.sort_values(by='input_order')
len(fetched_df)
```
```
57.9 ms ± 8.26 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

With  Index (3x faster) :

```python
df_with_index =  pg._vertex_prop_dataframe.set_index('_VERTEX_')
```
```python
%%timeit
fetched_df = df_with_index.loc[node_ids_to_fetch]
```
```
18.5 ms
```

### PR DESCRIPTION
**Pull request by eriknw**
~Currently, this only does SG version for #2401.  MG is still TODO.~

Closes #2401

This also doesn't change anything user-facing (yet).

### CODE REVIEW

**Comment by alexbarghi-nv** [comment]
@eriknw is this ready for review?

**Comment by eriknw** [comment]
Yeah, for SG.  I'm going to put MG into a different WIP branch, because of issues I'm having with `dask_cudf`.

I still want to add one thing: the ability to add vertex data that already has the vertex id as the index.

**Comment by alexbarghi-nv** [comment]
ok, I'll wait until you've made that addition, I wasn't sure if that was going into this branch or now

**Review by rlratzel — APPROVED** [review]
LGTM, I had just one optional minor question/suggestion which need not hold up approval.

File: python/cugraph/cugraph/structure/property_graph.py
Comment (eriknw) [reply python/cugraph/cugraph/structure/property_graph.py:452]: I doubt that it's any faster, but I aspire to write canonical pandas, and, who knows, maybe pandas can do something smarter and/or maybe cudf will add this API in the future.

**Comment by BradReesWork** [comment]
rerun tests

**Comment by alexbarghi-nv** [comment]
rerun tests

**Comment by BradReesWork** [comment]
rerun tests

**Review by VibhuJawa — CHANGES_REQUESTED** [review]
Can we add a check that we fetch ids in correct order and behave correctly with repeated ids ?

See example:
```python
import cudf
import cugraph
from cugraph.experimental import PropertyGraph
import numpy as np

src_ser = cudf.Series([1, 1, 1, 2])
dst_ser = cudf.Series([2, 3, 4, 1])
df = cudf.DataFrame(
    {"src": src_ser, "dst": dst_ser, "edge_feat": np.arange(len(src_ser))}
)
pg = PropertyGraph()
pg.add_edge_data(df, vertex_col_names=['src', 'dst'])
```
Current Main Behavior: 

```
pg.get_edge_data(edge_ids = [2,1,3,1], columns=['edge_feat'])

_SRC_	_DST_	_EDGE_ID_	_TYPE_	edge_feat
1	1	3	1		1
2	1	4	2		2
3	2	1	3		3
```

Expected behavior on this PR.  

```
_SRC_	_DST_	_EDGE_ID_	_TYPE_	edge_feat
0	1	4	2		2
1	1	3	1		1
2	2	1	3		3
3	1	3	1		1
```

File: python/cugraph/cugraph/tests/mg/test_mg_property_graph.py
Comment (VibhuJawa) [inline python/cugraph/cugraph/tests/mg/test_mg_property_graph.py:650]: Can we add a similar test for get_node_data_repeated too please as that goes through a different function  .

**Review by VibhuJawa — CHANGES_REQUESTED** [review]

**Comment by BradReesWork** [comment]
@gpucibot merge

File: python/cugraph/cugraph/tests/mg/test_mg_property_graph.py
Comment (eriknw) [reply python/cugraph/cugraph/tests/mg/test_mg_property_graph.py:650]: Test added in https://github.com/rapidsai/cugraph/pull/2510

**Pull request merged**
